server/src/main/java/org/opensearch/search/aggregations/bucket/composite/BinaryValuesSource.java

```diff
@@ -101,10 +104,9 @@ void copyCurrent(int slot) {
     @Override
     int compare(int from, int to) {
         if (missingBucket) {
-            if (values.get(from) == null) {
-                return values.get(to) == null ? 0 : -1 * reverseMul;
-            } else if (values.get(to) == null) {
-                return reverseMul;
+            int result = missingOrder.compare(() -> Objects.isNull(values.get(from)), () -> Objects.isNull(values.get(to)), reverseMul);
+            if (!MissingOrder.unknownOrder(result)) {
```

Comment:
nit; please use explicit boolean checks.. negation is sometimes hard to see on first pass.

```suggestion
            if (MissingOrder.unknownOrder(result) == false) {
```

Reply:
Done.